lab_3/main.py

```diff
@@ -20,7 +20,53 @@ def tokenize_by_sentence(text: str) -> tuple:
          (('_', 'h', 'e', '_'), ('_', 'i', 's', '_'), ('_', 'h', 'a', 'p', 'p', 'y', '_'))
          )
     """
-    pass
+    if not isinstance(text, str):
+        return ()
+
+    tokenized_sentence = ''
+    tokenized_sentences = []
+    underscore = ['_']
+
+    for symbol in text:
+        if symbol == 'ö':
+            text = text.replace(symbol, 'oe')
+        elif symbol == 'ü':
+            text = text.replace(symbol, 'ue')
+        elif symbol == 'ä':
+            text = text.replace(symbol, 'ae')
+        elif symbol == 'ß':
+            text = text.replace(symbol, 'ss')
+
+    special_symbols = ['`', '~', '!', '@', '#', '$', '%', '^', '&', '*', '(', ')', '_', '-', '+', '=', '{', '[', ']',
+                       '}', '|', '\\', ':', ';', '"', "'", '<', ',', '>', '.', '?', '/', '1', '2', '3', '4', '5', '6',
+                       '7', '8', '9', '0']
+
+    for symbol in special_symbols:
```

Comment:
The same goes for the umlauts - iterate over them like here, instead of iterating over the whole text.